docs/HowToSupportTizenRtProtectedBuild.md

```diff
@@ -0,0 +1,343 @@
+# Protected Build Porting Guide
+
+
+## Contents
+- [Introduction](#introduction)
+	- [Prerequisites](#prerequisites)
+	- [Overview of Protected Build](#overview-of-protected-build)
+- [Steps for suporting protected build in a new chipset](#steps-for-supporting-protected-build-in-a-new-chipset)
+	- [Creating a new configuration for Protected Build](#creating-a-new-configuration-for-protected-build)
+	- [Creating linker scripts](#creating-linker-scripts)
+	- [Userspace Initialization](#userspace-initialization)
+	- [MPU Initialization](#mpu-initialization)
+	- [Allocation of kernel and user heaps](#allocation-of-kernel-and-user-heaps)
+	- [Changes in Make files](#changes-in-make-files)
+	- [Changes in the download method](#changes-in-the-download-method)
+
+
+## Introduction
+This guide explains the steps required to support protected build on a new Chipset.
+
+### Prerequisites
+- TizenRT 3.0 flat build needs to be successfully ported and tested on the target CHIPSET board.
+- The generic code relating to TizenRT protected build should already be present in the codebase (Configuration variable: CONFIG_ BUILD_PROTECTED)
```

Comment:
I don't know what you mean and what I should prepare. What is the generic code?

Reply:
Here, we mean to say that TizenRT would already contain many pieces of code under CONFIG_ BUILD_PROTECTED configuration to support protected build. These will be generic and non chipset specific code. In this document we would be giving steps to include chipset and project specific code for protected build. However, if you feel that this point does not make sense, then we will remove it..

Reply:
I mean we should write exact information or steps. I don't want the word like `generic code`. The purpose of porting guide documentation is that let the beginner know how to do it. I don't think that beginner knows the meaning of `generic code` and what it is, what he should do.
In other words, let's change the `generic code` to `xxx.c file or yyy function or any specific code`. That's what I want.

Reply:
Ok. But the code relating to protected build is spread across many files. So, its not possible to specify the files separately. However, I think since the user would be using the latest tizenrt code, we can safely assume that the protected build code is already present in it. So, we dont have to explicitly mention this point. We will remove it.